internal/resolver/ooniresolver/ooniresolver.go

```diff
@@ -101,6 +101,17 @@ func (c *Resolver) newQueryWithQuestion(q dns.Question) (query *dns.Msg) {
 	query.RecursionDesired = true
 	query.Question = make([]dns.Question, 1)
 	query.Question[0] = q
+	// Add padding for DoH and DoT according to RFC8467
+	if c.Transport().RequiresPadding() {
+		query.SetEdns0(4096, true)
+		// Clients SHOULD pad queries to the closest multiple of 128 octets
+		// rfc8467#section-4.1
+		paddReminder := (128 - query.Len()) % 128
+		qPadding := new([128]byte)
+		qPaddingOpt := new(dns.EDNS0_PADDING)
+		qPaddingOpt.Padding = qPadding[0:paddReminder]
```

Comment:
Can this become

```Go
qPaddingOpt.Padding = new([paddReminder]byte)
```

?

Reply:
No :man_shrugging: 
```
cannot use new([paddReminder]byte) (value of type *[-1]byte) as []byte value in assignment
array length paddReminder (variable of type int) must be constant
```

Reply:
Right, I have suggested the wrong change, please try this:

```diff
diff --git a/internal/resolver/ooniresolver/ooniresolver.go b/internal/resolver/ooniresolver/ooniresolver.go
index 8680703..3c4a1d4 100644
--- a/internal/resolver/ooniresolver/ooniresolver.go
+++ b/internal/resolver/ooniresolver/ooniresolver.go
@@ -107,9 +107,8 @@ func (c *Resolver) newQueryWithQuestion(q dns.Question) (query *dns.Msg) {
                // Clients SHOULD pad queries to the closest multiple of 128 octets
                // rfc8467#section-4.1
                paddReminder := (128 - query.Len()) % 128
-               qPadding := new([128]byte)
                qPaddingOpt := new(dns.EDNS0_PADDING)
-               qPaddingOpt.Padding = qPadding[0:paddReminder]
+               qPaddingOpt.Padding = make([]byte, paddReminder)
                query.IsEdns0().Option = append(query.IsEdns0().Option, qPaddingOpt)
        }
        return
```

Reply:
(I don't have any other cosmetic comment after this and the next steps are roughly: (1) read the RFC and double check your implementation, (2) write unit tests, (3) merge.)